packages/flutter/lib/src/material/icon.dart

```diff
@@ -65,13 +83,13 @@ class Icon extends StatelessComponent {
       category = parts[0];
       subtype = parts[1];
     }
-    // TODO(eseidel): This clearly isn't correct.  Not sure what would be.
-    // Should we use the ios images on ios?
-    String density = 'drawable-xxhdpi';
+    // TODO(kgiesing): Should selector logic be moved down a level so that
+    // any generic asset can be resolved using consistent contextual logic?
```

Comment:
Well my point is that the values 0.875, 1.75, etc, and the strings "xxxhdpi", "ldpi", etc, should be at this layer and not below. But maybe a function which takes an ordered table of <double, T> and does a binary search on this table to find the T value that corresponds to the appropriate range is something that would be valuable at a layer below.